codice/ddf
# DDF-04774 Session timeout dialog goes into negatives

#### What does this PR do?
Fixes an issue where the client running on a web browser would go into the negatives if the server became disconnected. Also migrates session-timeout.view.js to a react component.
#### Who is reviewing it? 
@mojogitoverhere @willwill96 

#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler 
@djblue
@bdeining 
#### How should this be tested?
Start up DDF and navigate to search ui
shutdown the server 
wait for the session timeout window to appear
ensure the timeout doesn't go into the negatives
ensure the new message prompting the user to refresh comes up

#### Any background context you want to provide?
#### What are the relevant tickets?
For Jira:
[](https://codice.atlassian.net/browse/)

For GH Issues:
Fixes: #4774 

#### Screenshots
Session timeout timer
![image (3)](https://user-images.githubusercontent.com/36056781/57032246-ef848d80-6c17-11e9-8892-e53db9385018.png)
Session Expired
![image (4)](https://user-images.githubusercontent.com/36056781/57032260-f6ab9b80-6c17-11e9-9d52-d5f6e0bcfd35.png)


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
